src/test/kotlin/sirius/web/http/WebServerNightlyTest.kt

```diff
@@ -0,0 +1,134 @@
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import sirius.kernel.SiriusExtension
+import sirius.kernel.Tags
+import sirius.kernel.commons.Json
+import java.io.InputStream
+import java.net.HttpURLConnection
+import java.net.URI
+import kotlin.test.assertEquals
+
+/**
+ * Simulates a bunch of "real" (outside) requests through netty and sirius.
+ * <p>
+ * This ensures a basic performance profile and also makes sure no trivial race conditions or memory leaks
+ * are added.
+ *
+ * This contains long running tests to be executed nightly.
+ */
+@Tag(Tags.NIGHTLY)
+@ExtendWith(SiriusExtension::class)
+class WebServerNightlyTest {
+
+    /**
+     * Invoke a web dispatcher which previously blocked the event loop and crashed netty.
+     * <p>
+     * We now fork a thread for every request so that we never block the event loop
+     * in {@link sirius.web.http.Response#contentionAwareWrite(Object, boolean)} but always a worker thread.
+     * Therefore the event loop can shovel away the data in the output buffer of the channel
+     * and the future will eventually fullfilled.
+     */
+    @Test
+    fun `Invoke large-blocking-calls with GET`() {
+
+        val url = URI("http://localhost:9999/large-blocking-calls").toURL().openConnection() as HttpURLConnection
+
+        url.setRequestMethod("GET")
+        url.setDoOutput(false)
+
+        val counter = countBytesInStream(url.inputStream)
+
+        assertEquals(180000000, counter)
+    }
+
+    private fun countBytesInStream(input: InputStream): Int {
+        var counter = 0
+        var count: Int
+        val buffer = ByteArray(8192)
+        while ((input.read(buffer).also { count = it }) > 0) {
+            counter += count
+        }
+
+        return counter
+    }
+
+    @Test
+    fun `HTTP pipelining is supported correctly`() {
+
+        val responses = ArrayList<HttpResponse>()
+
+        val workerGroup = NioEventLoopGroup()
+
+        try {
+            val b = Bootstrap()
+            b.group(workerGroup)
+            b.channel(NioSocketChannel::class.java)
+            b.handler(object : ChannelInitializer<SocketChannel>() {
+                override fun initChannel(ch: SocketChannel) {
+                    ch.pipeline().addLast(HttpClientCodec())
+                    ch.pipeline().addLast(object : ChannelInboundHandlerAdapter() {
+                        override fun channelRead(ctx: ChannelHandlerContext, msg: Any) {
+                            if (msg is HttpResponse) {
+                                responses.add(msg)
+                            }
+                            super.channelRead(ctx, msg)
+                            if (responses.size == 3) {
+                                ctx.channel().close()
+                            }
+                        }
+                    })
+                }
+            })
+
+            val f = b.connect("localhost", 9999).sync()
```

Comment:
**Optional:** Rename `f` to something more descriptive, can't even guess the data type in this case 😅 Very good examply of why single letter variable names should be changed 👍🏻